src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -18,6 +22,37 @@ public void actions_implementing_DumbAware_should_extend_DumbAwareAction() {
         .check(importedClasses);
   }
 
+  @Test
+  public void actions_overriding_onUpdate_should_call_super_onUpdate() {
+    classes()
+        .that()
+        .areAssignableTo(com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.class)
+        .and()
+        .areNotAssignableFrom(com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.class)
+        .and(new DescribedPredicate<JavaClass>("override onUpdate method") {
+          @Override
+          public boolean apply(JavaClass input) {
+            return input.getMethods().stream().anyMatch(method -> method.getName().equals("onUpdate"));
```

Comment:
`get(Declared)Methods`, as opposed to `getAllMethods`, which would include the inherited ones as well